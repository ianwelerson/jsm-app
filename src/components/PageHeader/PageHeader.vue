<script setup lang="ts">
import { computed, reactive } from 'vue'
import IconBase from '@/components/IconBase/IconBase.vue'

interface PageHeaderState {
  menuStatus: MenuState | null
}

type MenuState = 'opened' | 'closed'

const state = reactive<PageHeaderState>({
  menuStatus: null,
})

const menuClasses = computed(() => {
  return [
    'header-menu',
    state.menuStatus ? `header-menu--${state.menuStatus}` : '',
  ]
})

const toggleMenu = () => {
  state.menuStatus = state.menuStatus === 'opened' ? 'closed' : 'opened'
}
</script>

<template>
  <header
    data-testid="page-header"
    class="page-header"
    :class="{ 'page-header--menu-opened': state.menuStatus === 'opened' }"
  >
    <div class="page-header__content">
      <RouterLink :to="{ name: 'appHome' }" class="header-logo">
        <img
          src="@/assets/images/logo.svg"
          alt="Juntos Somos Mais - Logo"
          class="header-logo__image"
        />
      </RouterLink>
      <div :class="menuClasses">
        <div class="header-menu__item" id="tp-header-search-spot"></div>
        <div class="header-menu__item">
          <div class="fake-content"></div>
        </div>
        <div class="header-menu__item">
          <div class="fake-content"></div>
        </div>
        <div class="close">
          <button
            @click="toggleMenu"
            class="close__btn"
            data-testid="page-header-menu-open"
          >
            <IconBase name="IconClose" :height="32" color="#222D39" />
          </button>
        </div>
      </div>
      <div class="mobile-menu">
        <button
          @click="toggleMenu"
          class="mobile-menu__btn"
          data-testid="page-header-menu-close"
        >
          <IconBase name="IconMenu" :height="32" color="#222D39" />
        </button>
      </div>
    </div>
  </header>
</template>

<style lang="scss" scoped>
.page-header {
  background-color: $gray-300;
  display: flex;
  justify-content: center;
  position: relative;
  z-index: 0;

  &__content {
    align-items: center;
    display: flex;
    justify-content: space-between;
    max-width: $content-desktop;
    margin: $spacing-6;
    width: 100%;

    @include screen('xl') {
      margin-right: 0;
      margin-left: 0;
    }
  }

  &--menu-opened {
    @include background-shadow;
  }
}

.header-menu {
  @include side-menu;

  @include screen('lg') {
    background: $transparent;
    box-shadow: none;
    flex: 1;
    justify-content: flex-end;
    flex-direction: row;
    height: auto;
    opacity: 1;
    padding: 0;
    position: relative;
    margin-left: $spacing-3;
    min-width: auto;
    transform: none;
    width: auto;
  }

  &__item {
    width: 100%;

    @include screen('lg') {
      max-width: 100%;
      width: auto;
    }

    &:not(:last-child) {
      margin-bottom: $spacing-3;

      @include screen('lg') {
        margin-bottom: 0;
        margin-right: $spacing-3;
      }
    }
  }
}

.header-logo {
  width: 148px;
  min-width: 148px;

  &__image {
    object-fit: contain;
  }
}

.mobile-menu {
  @include screen('lg') {
    display: none;
  }

  &__btn {
    background-color: $transparent;
    border: none;
  }
}

.fake-content {
  background-color: $gray-500;
  border-radius: $radius-extra;
  height: 20px;
  width: 100%;

  @include screen('lg') {
    height: 16px;
    width: 150px;
  }
}
</style>
