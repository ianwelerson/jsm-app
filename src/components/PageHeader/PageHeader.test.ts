import { render, waitFor } from '@testing-library/vue'
import PageHeader from './PageHeader.vue'

describe('PageHeader', () => {
  it('should render page header', async () => {
    const { getByTestId } = render(PageHeader)

    getByTestId('page-header')
  })

  it('should open and close mobile menu', async () => {
    const { getByTestId } = render(PageHeader)

    getByTestId('page-header-menu-open').click()

    await waitFor(() => {
      expect(
        getByTestId('page-header').classList.contains(
          'page-header--menu-opened'
        )
      ).toBeTruthy()
    })

    getByTestId('page-header-menu-close').click()

    await waitFor(() => {
      expect(
        getByTestId('page-header').classList.contains(
          'page-header--menu-opened'
        )
      ).toBeFalsy()
    })
  })

  it('should match the snapshot', async () => {
    const { html } = render(PageHeader)

    expect(html()).toMatchSnapshot()
  })
})
