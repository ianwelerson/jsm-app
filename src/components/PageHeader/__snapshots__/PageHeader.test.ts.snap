// Vitest Snapshot v1

exports[`PageHeader > should match the snapshot 1`] = `
"<header data-testid=\\"page-header\\" class=\\"page-header\\" data-v-40300573=\\"\\">
  <div class=\\"page-header__content\\" data-v-40300573=\\"\\">
    <routerlink to=\\"[object Object]\\" class=\\"header-logo\\" data-v-40300573=\\"\\"><img src=\\"/src/assets/images/logo.svg\\" alt=\\"Juntos Somos Mais - Logo\\" class=\\"header-logo__image\\" data-v-40300573=\\"\\"></routerlink>
    <div class=\\"header-menu\\" data-v-40300573=\\"\\">
      <div class=\\"header-menu__item\\" id=\\"tp-header-search-spot\\" data-v-40300573=\\"\\"></div>
      <div class=\\"header-menu__item\\" data-v-40300573=\\"\\">
        <div class=\\"fake-content\\" data-v-40300573=\\"\\"></div>
      </div>
      <div class=\\"header-menu__item\\" data-v-40300573=\\"\\">
        <div class=\\"fake-content\\" data-v-40300573=\\"\\"></div>
      </div>
      <div class=\\"close\\" data-v-40300573=\\"\\"><button class=\\"close__btn\\" data-testid=\\"page-header-menu-open\\" data-v-40300573=\\"\\">
          <div data-testid=\\"icon-base\\" class=\\"icon-base\\" data-v-2336b232=\\"\\" data-v-40300573=\\"\\">
            <!---->
          </div>
        </button></div>
    </div>
    <div class=\\"mobile-menu\\" data-v-40300573=\\"\\"><button class=\\"mobile-menu__btn\\" data-testid=\\"page-header-menu-close\\" data-v-40300573=\\"\\">
        <div data-testid=\\"icon-base\\" class=\\"icon-base\\" data-v-2336b232=\\"\\" data-v-40300573=\\"\\">
          <!---->
        </div>
      </button></div>
  </div>
</header>"
`;
